### ISSUE TITLE
ngFor TrackbyFunction support ng-template with ngTemplateOutlet.

### ISSUE THREAD

**Issue by xiexin36** [angular/angular#23406]
## I'm submitting a...

<pre><code>

[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Performance issue
[x] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
[ ] Other... Please describe:
</code></pre>
        <div *ngFor="let item of sourcelist; trackBy: TrackbyFunction" (click)="clickHandler(item)" style="margin:0px;">
            <ng-template [ngTemplateOutlet]="template" [ngTemplateOutletContext]="{item:item}">
            </ng-template>
        </div>
## Expected behavior

Above code, every time I set sourcelist, all the template will be generate again, even I have set trackBy, 
also I tried this https://angular.io/api/common/NgForOf sample, 
<pre><code>
<ng-template ngFor let-item [ngForOf]="items" let-i="index" [ngForTrackBy]="trackByFn">
  <li>...</li>
</ng-template>
</code></pre>
but it does not support ngTemplateOutlet, I need ngTemplateOutlet to get templte outside this component.

## Environment

<pre><code>
Angular version: X.Y.Z

Browser:
- [ X] Chrome (desktop) version XX
- [X ] Chrome (Android) version XX
- [ X] Chrome (iOS) version XX
- [X ] Firefox version XX
- [X ] Safari (desktop) version XX
- [X ] Safari (iOS) version XX
- [ X] IE version XX
- [ X] Edge version XX
 
For Tooling issues:
- Node version: XX  
- Platform:  

Others:

</code></pre>

**Comment by sandangel** [comment]
Hi, I think this is an expected behaviour because when your ref to the sourcelist change. If you don't want that, using push() or splice() to modify array and use ChangeDetectionStrategy.Default (IMO this is a bad practice though)

**Comment by xiexin36** [comment]
@sandangel Yes, I will try it, but just like you sad, it will be more complex.....
and @vicb Sorry, maybe my poor English didn't explain it clearly,  With this https://angular.io/api/common/NgForOf sample, it doesn't support ngTemplateOutlet, but I need to ref a outside component, just like a ListView can set a ItemTemplate.

**Comment by xiexin36** [comment]
And I also post it on StackOverflow here, https://stackoverflow.com/questions/49870294/ngfor-trackbyfunction-support-ng-template-with-ngtemplateoutlet
